packages/workbox-cache-expiration/CacheExpirationManager.mjs

```diff
@@ -145,7 +149,19 @@ class CacheExpirationManager {
    * @private
    */
   async _findExtraEntries() {
-    return [];
+    const extraUrls = [];
+
+    if (!this._maxEntries) {
+      return [];
+    }
+
+    const timestamps = await this._timestampModel.getAllTimestamps();
+    while (timestamps.length > this._maxEntries) {
```

Comment:
This would loop infinitely if a developer inadvertantly sets `this._maxEntries` to a negative number.